ui/app/AppLayouts/Chat/views/ChatColumnView.qml

```diff
@@ -186,19 +178,17 @@ Item {
                     delegate: Repeater {
                         model: subItems
                         delegate: ChatContentView {
+                            store: rootStore
                             Component.onCompleted: {
-                                parentModule.prepareChatContentModuleForChatId(model.itemId)
-                                chatContentModule = parentModule.getChatContentModule()
+                                rootStore.chatCommunitySectionModule.prepareChatContentModuleForChatId(model.itemId);
+                                rootStore.chatContentModule = rootStore.currentChatContentModule();
```

Comment:
@saledjenic this is basdically calling `rootStore.currentChatContentModule()` which is basically returning `chatCommunitySectionModule.getChatContentModule()`, what am I missing?

Reply:
@alexandraB99 the difference it significant cause `rootStore.currentChatContentModule()` will always return the same `chatContentModule` which will correspond to the currently selected chat/channel.

But having this in the delegate:
```
rootStore.chatCommunitySectionModule.prepareChatContentModuleForChatId(model.itemId)
chatContentModule = rootStore.chatCommunitySectionModule.getChatContentModule();
```
this sets for each item from a model appropriate `chatContentModule` based on the `model.itemId`

does this make sense to you now?

Reply:
Isn't the delegate the current activeItem any how? this exactly the reason I added these lines:
```
onChatItemSelected: {
    root.store.chatCommunitySectionModule.setActiveItem(id, "")         root.store.chatCommunitySectionModule.prepareChatContentModuleForChatId(id);
```

to set the active item as soon as the chat is selected and get rid of additional properties. Please let me know if I'm missing something here.

Reply:
> Isn't the delegate the current activeItem any how?

@alexandraB99  not sure that I understood your question completely, but delegate is instantiated for each item of the model (so each chat/channel represents a single delegate, when you click any of them calling `setActiveItem` we're setting appropriate chat/channel as active one and from that moment it may be accessed `chatCommunitySectionModule.activeItem` in case of chat or channel which doesn't belong to any category or via `chatCommunitySectionModule.activeItem.activeSubItem` in case of channel which belongs to some category.

in case of Chat section that should be:
```
onChatItemSelected: root.chatSectionModule.setActiveItem(id, "")
```

in case of Community section:
```
onChatItemSelected: {
if(categoryId === "")
    root.communitySectionModule.setActiveItem(id, "")
else
    root.communitySectionModule.setActiveItem(categoryId, id)
}
```

So no point of adding `root.store.chatCommunitySectionModule.prepareChatContentModuleForChatId(id);` in that context. Calling `prepareChatContentModuleForChatId` is actually needed only when we want to fetch chat content module (calling `getChatContentModule `).